comp/etw/impl/etwSession.go

```diff
@@ -211,3 +222,28 @@ func createEtwSession(name string) (*etwSession, error) {
 
 	return nil, fmt.Errorf("StartTraceW failed; %w", err)
 }
+
+func createWellKnownEtwSession(name string) (*etwSession, error) {
+	utf16SessionName, err := windows.UTF16FromString(name)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect session name; %w", err)
+	}
+
+	s := &etwSession{
+		Name:      name,
+		utf16name: utf16SessionName,
+		wellKnown: true,
+	}
+	sessionNameSize := (len(utf16SessionName) * int(unsafe.Sizeof(utf16SessionName[0])))
+	bufSize := int(unsafe.Sizeof(C.EVENT_TRACE_PROPERTIES{})) + sessionNameSize
+	propertiesBuf := make([]byte, bufSize)
+
+	pProperties := (C.PEVENT_TRACE_PROPERTIES)(unsafe.Pointer(&propertiesBuf[0]))
+	pProperties.Wnode.BufferSize = C.ulong(bufSize)
+	pProperties.Wnode.ClientContext = 1
+	pProperties.Wnode.Flags = C.WNODE_FLAG_TRACED_GUID
+
+	pProperties.LogFileMode = C.EVENT_TRACE_REAL_TIME_MODE
+	s.propertiesBuf = propertiesBuf
```

Comment:
In my own opinion it feels that it is waste of time or at least it is largely duplicated. I think the reason is that session contains a reference to the properties. Because it technically is not used anywhere should we not to persist properties across function call at all.